rfd/0227-ec2-discovery-by-aws-org.md

```diff
@@ -0,0 +1,442 @@
+                "Resource": [
+                    "*"
+                ]
+            }
+        ]
+    }
+  ```
+  
+  Trust Policy:
+  ```json
+    {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Effect": "Allow",
+                "Principal": {
+                    "Service": "ec2.amazonaws.com"
+                },
+                "Action": "sts:AssumeRole"
+            }
+        ]
+    }
+  ```
+</details>
+
+The user also needs to create an IAM Role in each Account, which will be assumed by the Role assigned to the Discovery Service (Role above):
+
+<details>
+  <summary>IAM Role to create in every Account</summary>
+
+  Policy:
+  ```json
+    {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Effect": "Allow",
+                "Action": [
+                    "ec2:DescribeInstances",
+                    "ssm:DescribeInstanceInformation",
+                    "ssm:GetCommandInvocation",
+                    "ssm:ListCommandInvocations",
+                    "ssm:SendCommand"
+                ],
+                "Resource": [
+                    "*"
+                ]
+            }
+        ]
+    }
+  ```
+  
+  Trust Policy:
+  ```json
+    {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Effect": "Allow",
+                "Principal": {
+                    "AWS": "<Role's ARN assigned to the Discovery Service>"
+                },
+                "Action": "sts:AssumeRole"
+            }
+        ]
+    }
+  ```
+</details>
+
+Still in step 2, the Alice is asked to add the `AmazonSSMManagedInstanceCore` IAM managed policy to the target EC2 instances.
+Alongside this policy, the user must also include the `organizations:DescribeOrganization` action:
+<details>
+  <summary>IAM Role to create in each Account</summary>
+
+  Policy:
+  ```json
+    {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Effect": "Allow",
+                "Action": [
+                    "organizations:DescribeOrganization"
+                ],
+                "Resource": [
+                    "*"
+                ]
+            }
+        ]
+    }
+  ```
+  
+  Trust Policy:
+  ```json
+    {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Effect": "Allow",
+                "Principal": {
+                    "Service": "ec2.amazonaws.com"
+                },
+                "Action": "sts:AssumeRole"
+            }
+        ]
+    }
+  ```
+</details>
+
+Going to step 3, Alice is now asked to create an SSM Document in every Account and Region that have EC2 instances.
+
+On Step 4, Alice must deploy a teleport Discovery Service in the organization's management account, with the following configuration:
+```yaml
+version: v3
+# ...
+discovery_service:
+  enabled: true
+  aws:
+   - types: ["ec2"]
+     account_ids: ["*"] # new field
```

Comment:
Would this support specific accounts in addition to a wildcard? So you could specify a different role to assume, for example, different label matchers for different accounts, etc.

Reply:
Yeah, you can still create multiple aws matchers:
```yaml
discovery_service:
  enabled: true
  aws:
   - types: ["ec2"]
     account_ids: ["*"]
     regions: ["us-east-1","us-west-1"]
     assume_role:
        role_arn: "arn:aws:iam::*:role/<role that exists in every account>"
   - types: ["ec2"]
     account_ids: ["123456789012", "123456789013"] # new field
     regions: ["eu-west-2"]
     labels:
       env: prod
     assume_role:
        role_arn: "arn:aws:iam::*:role/<role that exists in both accounts>"
```

I'll add another user story that covers this use case.